mlrun/artifacts/document.py

```diff
@@ -301,13 +297,19 @@ def to_langchain_documents(
                 metadata[self.METADATA_CHUNK_KEY] = str(idx)
                 doc = Document(
                     page_content=text,
-                    metadata=metadata,
+                    metadata=metadata.copy(),
                 )
                 results.append(doc)
         return results
 
     def collection_add(self, collection_id: str) -> None:
-        self.spec.collections.add(collection_id)
+        if collection_id not in self.spec.collections:
+            self.spec.collections[collection_id] = "1"
+            project = mlrun.get_or_create_project(mlrun.mlconf.default_project)
+            project.update_artifact(self)
```

Comment:
The artifact should not take care of updating itself in the DB. It's up to the user to update if needed through the relevant producer.
Also, this code will work in some cases, but will always assume you're registering the document through a project, while the user can log the document through the MLRun execution context (i.e. the run is the producer).

Reply:
will fix